modules/effects/src/effects_root_module.ts

```diff
@@ -1,18 +1,22 @@
 import { NgModule, Inject, Optional } from '@angular/core';
-import { StoreModule } from '@ngrx/store';
+import { StoreModule, Store } from '@ngrx/store';
 import { EffectsRunner } from './effects_runner';
 import { EffectSources } from './effect_sources';
 import { ROOT_EFFECTS } from './tokens';
 
+export const ROOT_EFFECTS_INIT = '@ngrx/store/root effects init' as '@ngrx/store/root effects init';
+
 @NgModule({})
 export class EffectsRootModule {
   constructor(
     private sources: EffectSources,
     runner: EffectsRunner,
+    store: Store<any>,
     @Inject(ROOT_EFFECTS) rootEffects: any[],
     @Optional() storeModule: StoreModule
   ) {
     runner.start();
+    store.dispatch({ type: ROOT_EFFECTS_INIT });
```

Comment:
I think this needs to be beneath the lines where you add the effects.

Reply:
Changed.
Why exactly :smile: ?

Reply:
Dispatching and registering the effects happens synchronously. Moving the dispatch to after registering the effects will give the effects a chance to listen for this new action type.